diff --git a/app/src/main/java/com/github/libretube/helpers/BackupHelper.kt b/app/src/main/java/com/github/libretube/helpers/BackupHelper.kt
index cc3cc0f74..0bbcec36d 100644
--- a/app/src/main/java/com/github/libretube/helpers/BackupHelper.kt
+++ b/app/src/main/java/com/github/libretube/helpers/BackupHelper.kt
@@ -34,8 +34,9 @@ object BackupHelper {
     @OptIn(ExperimentalSerializationApi::class)
     suspend fun createAdvancedBackup(context: Context, uri: Uri, backupFile: BackupFile) {
         try {
-            context.contentResolver.openOutputStream(uri)?.use { outputStream ->
+            context.contentResolver.openOutputStream(uri, "wt")?.buffered()?.use { outputStream ->
                 JsonHelper.json.encodeToStream(backupFile, outputStream)
+                outputStream.flush()
             }
             context.toastFromMainDispatcher(R.string.backup_creation_success)
         } catch (e: Exception) {
